### PR TITLE
Feature/create candidate model

### LINKED ISSUE #6: Create Candidate model
Create entity and its attributes.

### PR DESCRIPTION
**Pull request by polgodia**
Created Candidate Model and Tested Creation and Update

### CODE REVIEW

File: src/test/java/cat/udl/eps/softarch/academicrecruit/domain/CandidateTest.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package cat.udl.eps.softarch.academicrecruit.domain;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class CandidateTest {
```
Comment (rogargon) [inline src/test/java/cat/udl/eps/softarch/academicrecruit/domain/CandidateTest.java]: This class doesn't seem necessary, specially to test methods that are actually generated by external code (Lombok)

File: src/test/java/cat/udl/eps/softarch/academicrecruit/steps/UpdateCandidateStepDefs.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        this.stepDefs = stepDefs;
+        this.candidateRepository = candidateRepository;
+    }
+
+
+    @When("I change the name of the candidate with id {string} to {string}")
+    public void iChangeNameOfCandidateTo(String id, String name) throws Throwable {
+
+        Candidate candidate = new Candidate();
+        candidate.setId(Long.valueOf(id));
```
Comment (rogargon) [inline src/test/java/cat/udl/eps/softarch/academicrecruit/steps/UpdateCandidateStepDefs.java]: No need to set the ID if it is autogenerated by the persistence layer (database)

File: src/test/java/cat/udl/eps/softarch/academicrecruit/steps/UpdateCandidateStepDefs.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        candidate.setName(name);
+
+        newResourceUri = "/candidates/"+ id;
```
Comment (rogargon) [inline src/test/java/cat/udl/eps/softarch/academicrecruit/steps/UpdateCandidateStepDefs.java]: To generate the final ID, the CandidateRepository should be used to save the new instance. The return value is the persisted candidate with the final ID. You can then use getUri()

File: src/test/java/cat/udl/eps/softarch/academicrecruit/steps/UpdateCandidateStepDefs.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+        stepDefs.result = stepDefs.mockMvc.perform(
```
Comment (rogargon) [inline src/test/java/cat/udl/eps/softarch/academicrecruit/steps/UpdateCandidateStepDefs.java:44]: You should create the existing candidate in a separate step (Given there is a candidate with name "oldname"...) and then update the name with a different one in this step

File: src/test/resources/features/UpdateCandidate.feature [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  As a admin
+  I want to update a candidate
+
+  Background:
+    Given There is a registered administrator with username "admin" and password "password" and email "admin@sample.app"
+    And I login as "admin" with password "password"
+    And I create a candidate with name "name"
+
+  Scenario: Admin updates name
+    When I change the name of the candidate with id "1" to "lluc"
```
Comment (rogargon) [inline src/test/resources/features/UpdateCandidate.feature]: Better not to used fixed IDs in the steps

File: src/test/resources/features/DeleteCandidate.feature [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  I want to delete existing candidates
+
+  Background:
+    Given There is a registered administrator with username "admin" and password "password" and email "admin@sample.app"
+
+  Scenario: Delete existing candidate
+    Given There is a registered candidate with dni "dni"
+    And I login as "admin" with password "password"
+    When I delete an existing candidate with name "name" and dni "dni"
+    Then The response code is 201
```
Comment (rogargon) [inline src/test/resources/features/DeleteCandidate.feature]: 201 (CREATED) is not the expected outcome when deleting a resource

**Review by rogargon — CHANGES_REQUESTED** [review]

**Review by bergacat1 — APPROVED** [review]

**Pull request merged**